lib/facility.rb

```diff
@@ -1,14 +1,85 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(hash_info)
+    @name = hash_info[:name]
+    @address = hash_info[:address]
+    @phone = hash_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") && vehicle.registration_date == nil
+      @collected_fees += collect_fees(vehicle)
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Time.new
+    end
```

Comment:
Overall this method looks good. To refactor it, one line 21, I would have pushed the responsibility of changing the vehicle's attribute into the Vehicle class, so creating a method in the Vehicle class called `set_registration_date` that encapsulates this logic.